listener.go

```diff
@@ -225,33 +225,52 @@ func insert(tableName string, rows []common.Row, txn *sql.Tx) bool {
 	return true
 }
 
+func getValueListFromKeys(row common.Row, pkeys []string) []interface{} {
+	var values []interface{}
+	// TODO Handle multiple data types
+	for _, pkey := range pkeys {
+		if row[pkey] == nil {
+			values = append(values, nil)
+		} else {
+			values = append(values, row[pkey].Value)
+		}
+	}
+	return values
+}
+
 func delete(tableName string, rows []common.Row, txn *sql.Tx) bool {
 	pkeys, err := getPkeysForTable(tableName)
+	sort.Strings(pkeys)
 	if len(pkeys) == 0 || err != nil {
-		log.Errorf("DELETE No primary keys found for table.", tableName)
+		log.Errorf("DELETE No primary keys found for table. %s", tableName)
+		return false
+	} else if len(rows) == 0 {
+		log.Errorf("No rows found for table.", tableName)
 		return false
 	} else {
-		sql := buildDeleteSql(tableName, pkeys)
+
+		sql := buildDeleteSql(tableName, rows[0], pkeys)
 		prep, err := txn.Prepare(sql)
 		if err != nil {
 			log.Errorf("DELETE Fail to prep statement [%s] error=[%v]", sql, err)
 			return false
 		}
 		defer prep.Close()
 		for _, row := range rows {
-			var values []interface{}
-			for _, pkey := range pkeys {
-				var value interface{}
-				row.Get(pkey, &value)
-				values = append(values, value)
-			}
-			_, err = prep.Exec(values)
-
+			values := getValueListFromKeys(row, pkeys)
+			// delete prepared statement from existing template statement
+			res, err := txn.Stmt(prep).Exec(values...)
 			if err != nil {
 				log.Errorf("DELETE Fail [%s] value=[%v] error=[%v]", sql, values, err)
 				return false
 			} else {
-				log.Debugf("DELETE Success [%s] value=[%v]", sql, values)
+				affect, err := res.RowsAffected()
+				if err == nil && affect != 0 {
```

Comment:
lets check `affect` separately, so that we can log a message like "No rows affected, indicating bad primary key" or something like that.  Also, can we rename `affect` to `affected` or `numRowsAffected`